extensions/arc/src/utils.ts

```diff
@@ -91,3 +91,12 @@ export class PermissionError extends Error {
 export const isPermissionError = (error: unknown) => {
   return error instanceof Error && error.name === "PermissionError";
 };
+
+export async function createNewArcWindow() {
+  await runAppleScript(`
+    tell application "Arc"
+      make new window
+      activate
+    end tell
```

Comment:
Can we somehow set the URL of the new tab here instead of using the `exec` command to open Arc. Later feels a bit like a hack.

Reply:
I tried a few variations of that approach originally but couldn't get the command to work reliably. However, I have limited AppleScript experience and I'm open to suggestions. I agree the `exec` approach isn't ideal 😕

Reply:
Hey @thomaspaulmann ,

I have been struggling with the same issue regarding `AppleScript` opening a new Arc tab. Using the default `Chromium` `AppleScript` integration for the `tab` dictionary` entry results in the browser crashing when attempting to open the new tab, which is what I imagine @zach-fuller is referring to, akin to attempting to do [this](https://github.com/raycast/extensions/blob/b3f7177eb0875c538cd6ef18737984e3ffb8afe0/extensions/browser-history/src/actions/index.ts#L39). I think it is crashing as they are attempting to do some advanced tab handling with their `Spaces` concept which they have not covered for the `AppleScript` path perhaps.

In any case, I have also reached the same conclusion, albeit not ideal, to open a new `Arc` tab, although my approach still leverages using `AppleScript` for it (which adds a layer of (redirection)[https://github.com/raycast/extensions/pull/3918/commits/30310cf45dc66b48109b230721ee768e0f9ceec4]), in hopes of being able to quickly move it back once this issue is fixed in Arc :).

Reply:
Well okay, if Arc crashes and this works, we should keep it. Thanks for the collaboration here. I'll reach out to the Arc team to see if they can fix the AppleScript on their side.

Reply:
Sorry about this crash folks! Have some changes to fix this crash, and provide a richer AppleScript interface into Arc so you don't have to do the `exec` gnarliness. Aiming to get that into the first release of 2023!